hngprojects/hng_boilerplate_python_fastapi_web
# feat: create a delete all testimonials endpoint

## Description
This PR introduces the following changes:

- Fixed DELETE Endpoint: Improved the DELETE endpoint at `/testimonials/` to correctly delete all testimonial entries from the database.
- Pytest: Created pytest to test the newly added endpoint.

**Related Issue**
#270 [FEAT]: API Endpoint to Delete All Testimonials

**Motivation and Context**
- The DELETE endpoint provides a way to clear all testimonials, which is useful for maintenance or testing purposes. Fixing the model warnings ensures that pytest can correctly collect and test the model without issues.

**How Has This Been Tested?**
- **Manual Testing:**
-Tested the DELETE endpoint to ensure it correctly removes all testimonials from the database.

-**Automated Testing:**

- Updated and ran unit tests using pytest to ensure all new and existing tests pass.
- Validated that the Testimonial model no longer triggers pytest warnings.

**The endpoint**
- Route `DELETE api/v1/testimonials`
- Method: DELETE
- Parameters: None

**DELETE Request Example:**

- Request URL: http://localhost:8000/testimonials/
- Response: {"message": "All testimonials deleted successfully", "data": {}, "status_code": 200}

**Types of changes**

**Checklist:**
